apps/aecore/test/aecore_chain_state_test.exs

```diff
@@ -10,41 +10,76 @@ defmodule AecoreChainStateTest do
   alias Aecore.Structures.TxData, as: TxData
   alias Aecore.Structures.SignedTx, as: SignedTx
   alias Aecore.Chain.ChainState, as: ChainState
+  alias Aecore.Chain.Worker, as: Chain
 
   test "block state" do
     block = get_block()
+    lock_time_block = Enum.at(block.txs, 0).data.lock_time_block
+    latest_block = Chain.latest_block()
 
-    assert %{"a" => %{balance:  1, nonce: 102},
-             "b" => %{balance: -2, nonce: 1},
-             "c" => %{balance: -3, nonce: 1}} ==
-      ChainState.calculate_block_state(block.txs)
+    assert %{"a" => %{balance: -9, nonce: 102,
+                      locked: [%{amount: 10, block: lock_time_block}]},
+             "b" => %{nonce: 1, balance: -11,
```

Comment:
I see that's the same as before, but why do we actually get negative balances?

Reply:
whoops, good question

Reply:
Because amounts are spent this block, but the received amounts for each are locked for one block? Correct me if I am wrong

Reply:
@cytadela8 this is just the block_state indicating the change this block makes to the chain_state, negative values occur for every token spent.

Reply:
Ohh, right. This is not the chain_state after the block. It's delta chain_state.